web/ui/react-app/src/pages/graph/Panel.tsx

```diff
@@ -138,55 +144,72 @@ class Panel extends Component<PanelProps, PanelState> {
         throw new Error('Invalid panel type "' + this.props.options.type + '"');
     }
 
-    fetch(`${this.props.pathPrefix}/${API_PATH}/${path}?${params}`, {
-      cache: 'no-store',
-      credentials: 'same-origin',
-      signal: abortController.signal,
-    })
-      .then(resp => resp.json())
-      .then(json => {
-        if (json.status !== 'success') {
-          throw new Error(json.error || 'invalid response JSON');
-        }
-
-        let resultSeries = 0;
-        if (json.data) {
-          const { resultType, result } = json.data;
-          if (resultType === 'scalar') {
-            resultSeries = 1;
-          } else if (result && result.length > 0) {
-            resultSeries = result.length;
-          }
+    let query;
+    let exemplars;
```

Comment:
As a side note, we should really start typing these responses, but that's not your job here :)

Reply:
I'd be down to work on that, just in different PR.